gt/main.c

```diff
@@ -1987,3 +2004,68 @@ l_update_gt_lua_states(lua_State *l)
 
 	return 0;
 }
+
+int
+l_update_gt_lua_states_incrementally(lua_State *l)
+{
+	int i;
+	uint32_t ctypeid;
+	struct gt_config *gt_conf;
+	uint32_t correct_ctypeid_gt_config = luaL_get_ctypeid(l,
+		CTYPE_STRUCT_GT_CONFIG_PTR);
+	const char *lua_filename;
+	const char *target_lua_filename;
+
+	/* First argument must be of type CTYPE_STRUCT_GT_CONFIG_PTR. */
+	void *cdata = luaL_checkcdata(l, 1,
+		&ctypeid, CTYPE_STRUCT_GT_CONFIG_PTR);
+	if (ctypeid != correct_ctypeid_gt_config)
+		luaL_error(l, "Expected `%s' as first argument",
+			CTYPE_STRUCT_GT_CONFIG_PTR);
+
+	gt_conf = *(struct gt_config **)cdata;
+
+	/* Second and third arguments must be Lua file names. */
+	lua_filename = luaL_checkstring(l, 2);
+	target_lua_filename = luaL_checkstring(l, 3);
+
+	if (lua_gettop(l) != 3)
+		luaL_error(l, "Expected three arguments, however it got %d arguments",
+			lua_gettop(l));
+
+	lua_getglobal(l, "dylib");
+	lua_getfield(l, -1, "luac_func");
```

Comment:
We shouldn't need `luac_func()`. Take a look at the source of `luac(1)`; it does everything in C.

Reply:
After checking the source code of ``luac(1)`` and [string.dump()](https://www.lua.org/source/5.1/lstrlib.c.html#str_dump),  they are using the same Lua internal function call ``luaU_dump()`` to dump the Lua chunk.